application/controllers/ThemeOptionsController.php

```diff
@@ -594,17 +587,29 @@ public function actionUninstall() : void
     public function actionReset(int $gsid) : void
     {
         $templatename = App()->request->getPost('templatename');
-        if (Permission::model()->hasGlobalPermission('templates', 'update')) {
-            TemplateConfiguration::uninstall($templatename);
-            TemplateManifest::importManifest($templatename);
-            App()->setFlashMessage(sprintf(gT("The theme '%s' has been reset."), $templatename), 'success');
-            $this->redirect(array("themeOptions/index"));
-        } else {
-            App()->setFlashMessage(gT("We are sorry but you don't have permissions to do this."), 'error');
-            $this->redirect(
-                array("/admin/surveysgroups/sa/update/", 'id'=>$gsid)
+        if (!Permission::model()->hasGlobalPermission('templates', 'update')
+            && !Permission::model()->hasSurveysInGroupPermission($gsid, 'surveys', 'update')
```

Comment:
&& or || ?

Reply:
Sorry, it's correct. BUT: Not minimal change required for the new feature... ^^

Reply:
hasSurveysInGroupPermission didn't exist before, then : seems needed : part of minimal update

Reply:
Oh : git it …

Doing if(No perm) { OUT } it's really more clear than if (PERM) { A lot of line … } else {OUT}

Reply:
Yes.